src/application-navigation/ApplicationNavigation.jsx

```diff
@@ -0,0 +1,193 @@
+   * is provided, no NavigationPrompts are resolved when navigation items are selected.
+   */
+  disablePromptsForNavigationItems: PropTypes.bool,
+  /**
+   * A configuration object with information specifying the creation of the Extension buttons rendered within the
+   * ApplicationNavigation header.
+   */
+  extensionItems: extensionItemsPropType,
+  /**
+   * An element to render within the ApplicationNavigation menu, shifted to the drawer at the `medium` and below.
+   */
+  hero: PropTypes.element,
+  /**
+   * An array of configuration objects with information specifying the creation of navigation items. These items
+   * are rendered within the ApplicationNavigation header at larger breakpoints and within the drawer menu at smaller breakpoints.
+   */
+  navigationItems: navigationItemsPropType,
+  /**
+   * The Object (or function that returns an Object) that specifies the messages
+   * used to prompt the user when navigation items are selected while NavigationPrompts
+   * are rendered by the ApplicationNavigation content.
+   */
+  navigationPromptResolutionOptions: navigationPromptResolutionOptionsShape,
+  /**
+   * Key/Value pairs associating a string key entry to a Number notification count. The keys must correspond to a
+   * navigationItem or extensionItem key provided through their associated props.
+   */
+  notifications: PropTypes.object, // eslint-disable-line react/no-unused-prop-types
+  /**
+   * Callback function triggered on Drawer Menu state change
+   */
+  onDrawerMenuStateChange: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of an extensions item.
+   * Ex: `onSelectExtensionsItem(String selectedUtilityItemKey, Object metaData)`
+   */
+  onSelectExtensionItem: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Help utility item.
+   * If `onSelectHelp` is not provided, the Help utility item will not be rendered.
+   * Ex: `onSelectLogout()`
+   */
+  onSelectHelp: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Logout action button.
+   * If `onSelectLogout` is not provided, the Logout action button will not be rendered.
+   * Ex: `onSelectLogout()`
+   */
+  onSelectLogout: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of a navigation item.
+   * Ex: `onSelectNavigationItem(String selectedNavigationItemKey, Object metaData)`
+   */
+  onSelectNavigationItem: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of the Settings utility item.
+   * If `onSelectSettings` is not provided, the Settings utility item will not be rendered.
+   * Ex: `onSelectSettings()`
+   */
+  onSelectSettings: PropTypes.func,
+  /**
+   * A function to be executed upon the selection of a custom utility item.
+   * Ex: `onSelectUtilityItem(String selectedUtilityItemKey, Object metaData)`
+   */
+  onSelectUtilityItem: PropTypes.func,
```

Comment:
I'm noticing some of these callback are shortened to `onSelectHelp` or `onSelectionSettings`, where some are `onSelectNavigationItem` or `onSelectUtilityItem`.  Would it make sense to match these?

Reply:
All these props are carried forward from the terra-application-navigation package. I'd prefer not to update the existing props unless there's something egregious wrong with them.